keps/sig-cluster-lifecycle/wgs/1815-componentconfig-objectmeta/README.md

```diff
@@ -0,0 +1,696 @@
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals and non-goals of
+this KEP.  Describe why the change is important and the benefits to users.  The
+motivation section can optionally provide links to [experience reports][] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+The initial implementation of the ComponentConfig API's didn't include OjectMeta fields. The Kubernetes project tooling relies on using Kustomize with manually added ObjectMeta fields in the ComponentConfig manifests. This functionality was broken as strict decoders were implemented for ComponentConfig API's and forced a leninent path to be added for the current ComponentConfig API versions. Adding ObjectMeta will allow for valid usage with Kustomize and strict decoders for current and future ComponentConfig API's.
+
+Related issues for context:
+
+- https://github.com/kubernetes/kubeadm/issues/1727
+
+### Goals
+
+<!--
+List the specific goals of the KEP.  What is it trying to achieve?  How will we
+know that this has succeeded?
+-->
+
+- Add a standard set of ObjectMeta fields (name) to the ComponentConfig API's.
+  - kubelet
+  - kube-proxy
+  - kube-scheduler
+  - kube-controller-manager
+  - kube-apiserver
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP?  Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+- Graduate the API versions
+  - We will implement ObjectMeta within the current version of each ComponetConfig API
+- Implement every possible ObjectMeta field for ComponentConfig API's
+  - As these configs are singular per component, per cluster, they won't have or need the full suite of ObjectMeta similar to core Kubernetes API's (ie. Pods, Servcies, etc.)
+
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation.  The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+- for component in [kubelet kubeproxy kubecontrollermanager kubeapiserver kubescheduler]
+  - Add the following ObjectMeta fields
+    - name
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal and how do we mitigate.  Think broadly.
+For example, consider both security and how this will impact the larger
+kubernetes ecosystem.
+
+How will security be reviewed and by whom?
+
+How will UX be reviewed and by whom?
+
+Consider including folks that also work outside the SIG or subproject.
+-->
+
+Minimal. The ObjectMeta schema is standardized throughout the Kubernetes codebase and is already compatible with approaches codified by Kustomize.
```

Comment:
Can also add that _not_ implementing this proposal creates a risk of breaking tooling when we move to strict deserializers by default.